source/commoncrypto/commoncrypto_ecc.c

```diff
@@ -0,0 +1,607 @@
+    }
+
+    CFDataRef hash_ref =
+        CFDataCreateWithBytesNoCopy(cc_key->cf_allocator, message->ptr, message->len, kCFAllocatorNull);
+    CFDataRef signature_ref =
+        CFDataCreateWithBytesNoCopy(cc_key->cf_allocator, signature->ptr, signature->len, kCFAllocatorNull);
+
+    CFErrorRef error = NULL;
+
+    bool verified = SecKeyVerifySignature(
+        cc_key->pub_key_ref, kSecKeyAlgorithmECDSASignatureDigestX962, hash_ref, signature_ref, &error);
+
+    CFRelease(signature_ref);
+    CFRelease(hash_ref);
+
+    return verified ? AWS_OP_SUCCESS : aws_raise_error(AWS_ERROR_CAL_SIGNATURE_VALIDATION_FAILED);
+}
+
+static int s_derive_public_key_fn(struct aws_ecc_key_pair *key_pair) {
+    /* we already have a public key, just lie and tell them we succeeded */
+    if (key_pair->pub_x.buffer && key_pair->pub_x.len) {
+        return AWS_OP_SUCCESS;
+    }
+
+    return aws_raise_error(AWS_ERROR_UNSUPPORTED_OPERATION);
+}
+
+static void s_destroy_key_fn(struct aws_ecc_key_pair *key_pair) {
+    struct commoncrypto_ecc_key_pair *cc_key = key_pair->impl;
+
+    if (cc_key->pub_key_ref) {
+        CFRelease(cc_key->pub_key_ref);
+    }
+
+    if (cc_key->priv_key_ref) {
+        CFRelease(cc_key->priv_key_ref);
+    }
+
+    aws_byte_buf_clean_up_secure(&key_pair->key_buf);
+    aws_mem_release(key_pair->allocator, cc_key);
+}
+
+static struct aws_ecc_key_pair_vtable s_key_pair_vtable = {
+    .sign_message_fn = s_sign_message_fn,
+    .signature_length_fn = s_signature_length_fn,
+    .verify_signature_fn = s_verify_signature_fn,
+    .derive_pub_key_fn = s_derive_public_key_fn,
+    .destroy_fn = s_destroy_key_fn,
+};
+
+static struct commoncrypto_ecc_key_pair *s_alloc_pair_and_init_buffers(
+    struct aws_allocator *allocator,
+    enum aws_ecc_curve_name curve_name,
+    const struct aws_byte_cursor *pub_x,
+    const struct aws_byte_cursor *pub_y,
+    const struct aws_byte_cursor *priv_key) {
+    struct commoncrypto_ecc_key_pair *cc_key_pair =
+        aws_mem_calloc(allocator, 1, sizeof(struct commoncrypto_ecc_key_pair));
+
+    if (!cc_key_pair) {
+        return NULL;
+    }
+
+    cc_key_pair->cf_allocator = aws_wrapped_cf_allocator_new(allocator);
+
+    size_t s_key_coordinate_size = s_key_coordinate_byte_size_from_curve_name(curve_name);
+
+    if (!s_key_coordinate_size) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        goto error;
+    }
+
+    if ((pub_x && pub_x->len != s_key_coordinate_size) || (pub_y && pub_y->len != s_key_coordinate_size) ||
+        (priv_key && priv_key->len != s_key_coordinate_size)) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        goto error;
+    }
+
+    size_t total_buffer_size = s_key_coordinate_size * 3 + 1;
+
+    if (aws_byte_buf_init(&cc_key_pair->key_pair.key_buf, allocator, total_buffer_size)) {
+        goto error;
+    }
+
+    aws_byte_buf_secure_zero(&cc_key_pair->key_pair.key_buf);
```

Comment:
trivial: no need to secure-zero a buffer you just initialized, there's nothing in there
AWS_ZERO_ARRAY() or memset is sufficient